Backtesting/collectors/exchanges/exchange_history_collector/history_collector.py

```diff
@@ -147,17 +158,27 @@ async def get_ohlcv_history(self, exchange, symbol, time_frame):
                                                                  since=(since + (time_frame_sec * 1000)))
                 last_candle_timestamp = candles[-1][commons_enums.PriceIndexes.IND_PRICE_TIME.value]
 
-            if self.end_timestamp is not None:
-                while candles[-1][commons_enums.PriceIndexes.IND_PRICE_TIME.value] > self.end_timestamp:
-                    candles.pop(-1)
+                if candles:
+                    last_candle_timestamp = candles[-1][commons_enums.PriceIndexes.IND_PRICE_TIME.value]
+
+                    if self.end_timestamp is not None:
+                        while candles[-1][commons_enums.PriceIndexes.IND_PRICE_TIME.value] > self.end_timestamp:
+                            candles.pop(-1)
+                    self.exchange.uniformize_candles_if_necessary(candles)
+                    await self.save_ohlcv(exchange=exchange,
+                                  cryptocurrency=self.exchange_manager.exchange.get_pair_cryptocurrency(symbol),
+                                  symbol=symbol, time_frame=time_frame, candle=candles,
+                                  timestamp=[candle[0] + time_frame_sec for candle in candles], multiple=True)
+                    candles.clear()
```

Comment:
the idea here is to avoid writing too many data at once ?

Reply:
yes, right

Reply:
and there is no issue in making multiple inserts like this ? I mean no double saved candles etc ?

Reply:
It does exactly the same thing. 
Unit test is ok. 
Do you want a unit test with larger daterange on original version to compare with it ?

Reply:
Right, it tests are ok, it must be fine !